src/binaryen-c.cpp

```diff
@@ -732,6 +732,40 @@ BinaryenExpressionRef BinaryenUnreachable(BinaryenModuleRef module) {
 
   return static_cast<Expression*>(ret);
 }
+BinaryenExpressionRef BinaryenAtomicLoad(BinaryenModuleRef module, uint32_t bytes, uint32_t offset, BinaryenType type, BinaryenExpressionRef ptr) {
+  auto* ret = ((Module*)module)->allocator.alloc<Load>();
+
+  if (tracing) {
+    auto id = noteExpression(ret);
+    std::cout << "  expressions[" << id << "] = BinaryenAtomicLoad(the_module, " << bytes << ", " << offset << ", " << type << ", expressions[" << expressions[ptr] << "]);\n";
+  }
+  ret->isAtomic = true;
```

Comment:
it would be better to use a `Builder` for these, see e.g. the impl of `BinaryenAtomicWait`.